presto-thrift-interface/src/main/java/com/facebook/presto/thrift/interfaces/client/ThriftPrestoClient.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.thrift.interfaces.client;
+
+import com.facebook.swift.service.ThriftException;
+import com.facebook.swift.service.ThriftMethod;
+import com.facebook.swift.service.ThriftService;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import javax.annotation.Nullable;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Presto Thrift service definition.
+ * This thrift service needs to be implemented on the Thrift Node in order to be used with Thrift Connector.
+ */
+@ThriftService("presto-thrift")
+public interface ThriftPrestoClient
+        extends Closeable
+{
+    /**
+     * Returns definitions for session properties supported by Thrift Node.
+     * Session properties which are different from their default values will be passed to Thrift Node in selected method calls.
+     * This method is called once on Thrift Connector start up.
+     *
+     * @return definitions for session properties supported by Thrift Node.
+     */
+    @ThriftMethod(exception = @ThriftException(type = ThriftServiceException.class, id = 1))
+    List<ThriftPropertyMetadata> listSessionProperties();
+
+    /**
+     * Returns supported schema names.
+     *
+     * @return supported schema names.
+     */
+    @ThriftMethod(exception = @ThriftException(type = ThriftServiceException.class, id = 1))
+    List<String> listSchemaNames();
+
+    /**
+     * Returns tables for the given schema name.
+     *
+     * @param schemaNameOrNull schema name or {@literal null} value.
+     * @return a list of table names with corresponding schemas. If schema name is null then returns
+     * a list of tables for all schemas. Returns an empty list of a schema doesn't exist.
+     */
+    @ThriftMethod(exception = @ThriftException(type = ThriftServiceException.class, id = 1))
+    List<ThriftSchemaTableName> listTables(@Nullable String schemaNameOrNull);
```

Comment:
This is the exact translation of `ConnectorMetadata.listTables()` SPI method. Why should its thrift signature be different?